### PR TITLE
improve elementwise_add_mkldnn_op test code coverage

### PR DESCRIPTION
**Pull request by wojtuss**
This PR improves the `elementwise_add` MKL-DNN kernel's test code coverage.

test=develop

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py [hunk 1]
```diff
@@ -15,121 +15,318 @@
...
+    def init_axis(self):
+        self.axis = 1
 
 
-class TestMKLDNNElementwiseAddOp_rowwise_add_1(
-        TestElementwiseAddOp_rowwise_add_1):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
+@skip_check_grad_ci(
+    reason="[skip shape check] Use y_shape(1) to test broadcast.")
```
Comment (Sand3r-) [inline python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py]: I'm not sure what is meant here by "Use y_shape(1) to test broadcast".

Comment (luotao1) [reply python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py]: @zhupengyang Could you make the reason more readable?

**Review by Sand3r- — DISMISSED** [review]
LGTM with a few remarks:
1. In the future in terms of refactoring I recommend the removal of explicit fallback mechanism in elementwise_mkldnn_op in a similar fashion to how it was done in elementwise_mul. This way the broadcast tests can be simply removed.
2. To optimize for UT execution speed, it might be considered to decrease the size of convolution filters to ensure faster processing.
3. The message for skipping the CI check for small shape could be slightly improved as @luotao1 has already mentioned.

Edit: @luotao1 Github says that I've deleted the comment you wrote, but I did no such thing. Looks like a bug to me.

The deleted message said:
"@zhupengyang Could you make the reason more readable?"

File: python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py [hunk 2]
```diff
@@ -15,121 +15,318 @@
...
+    def init_axis(self):
+        self.axis = 1
 
-class TestMKLDNNElementwiseAddOp_broadcast_4(TestElementwiseAddOp_broadcast_4):
-    def init_kernel_type(self):
-        self.use_mkldnn = True
 
+class TestMKLDNNElementwiseAddOp_broadcast_4(TestMKLDNNElementwiseAddOp):
+    def print_info(self):
+        print("### TestMKLDNNElementwiseAddOp_broadcast_4 ###")
```
Comment (lidanqing-vv) [inline python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py]: Where is print_info function used? Others look good to me.

**Review by lidanqing-vv — COMMENTED** [review]
Is logging function needed in UT now?

File: python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py (hunk 2 above)
Comment (wojtuss) [reply python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py]: It is a leftover. I will remove it.

**Comment by wojtuss** [comment]
@Sand3r- , true,  I have refactored the code so the there is no fallback to the native implementation from the DNNL-based kernel anymore.

**Review by Sand3r- — DISMISSED** [review]
That looks much simpler. Thank you so much for the update!

File: python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py (hunk 2 above)
Comment (wojtuss) [reply python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py]: Done.

**Review by Sand3r- — APPROVED** [review]

**Review by luotao1 — APPROVED** [review]
LGTM
http://ci.paddlepaddle.org/viewLog.html?buildId=274504&tab=buildLog&buildTypeId=YYTest_PrCiCpuPy2&logTab=tail
```
[10:51:29]	[Step 1/1] python/paddle/fluid/tests/unittests/mkldnn/test_elementwise_add_mkldnn_op.py
[10:51:29]	[Step 1/1] -@skip_check_grad_ci(
[10:51:29]	[Step 1/1] -@skip_check_grad_ci(
```
@zhangting2020 Should we monitor the delete of `@skip_check_grad_ci` as well?

**Pull request merged**